compiler/rustc_lint_defs/src/lib.rs

```diff
@@ -929,9 +940,7 @@ macro_rules! impl_lint_pass {
     ($ty:ty => [$($lint:expr),* $(,)?]) => {
         impl $crate::LintPass for $ty {
             fn name(&self) -> &'static str { stringify!($ty) }
-        }
-        impl $ty {
-            pub fn get_lints() -> $crate::LintVec { vec![$($lint),*] }
+            fn get_lints(&self) -> $crate::LintVec { vec![$($lint),*] }
         }
```

Comment:
Can we keep? This will limit the churn of constructing the pass types.

Reply:
Still relevant.

Reply:
The trait resolver is having problems with this one, they'd need to have different names.